browser/brave_ads/creatives/search_result_ad/creative_search_result_ad_tab_helper.cc

```diff
@@ -166,12 +181,7 @@ void CreativeSearchResultAdTabHelper::MaybeHandleCreativeAdViewedEvents(
 void CreativeSearchResultAdTabHelper::MaybeHandleCreativeAdViewedEvent(
     const std::string& placement_id) {
   CHECK(!placement_id.empty());
-
-  // TODO(https://github.com/brave/brave-browser/issues/39404): Remove
-  // `base::Contains` and Sanitize JSON-LD placement id.
-  if (base::Contains(placement_id, '\"')) {
-    return;
-  }
+  CHECK(base::ranges::all_of(placement_id, &IsUnreservedCharOrPercentageSign));
```

Comment:
If the placement id on search.brave.com becomes malformed, would this not crash the browesr?

Reply:
Do we need this `CHECK`, because `base::EscapeAllExceptUnreserved` function would be tested.

Reply:
Replaced CHECK with following comment:

```
  // It is safe to pass placement_id directly to the JavaScript function as it
  // has all characters except alphanumerics and -._~ escaped.

```